src/openrpc/lifecycle.json

```diff
@@ -1,261 +1,494 @@
+              }
+            }
+          }
+        ]
+      },
+      {
+        "name": "onResume",
+        "tags": [
+          {
+            "name": "event"
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the suspended event",
+        "params": [],
+        "result": {
+          "name": "value",
+          "schema": {
+            "$ref": "#/components/schemas/LifecycleEvent"
+          }
+        },
+        "examples": [
+          {
+            "name": "Default Example",
+            "params": [],
+            "result": {
+              "name": "Default Result",
+              "value": {
+                "state": "paused",
+                "previous": "suspended"
+              }
+            }
+          }
+        ]
+      },
+      {
+        "name": "onHibernate",
+        "tags": [
+          {
+            "name": "event"
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the hibernating event",
+        "params": [],
+        "result": {
+          "name": "value",
+          "schema": {
+            "$ref": "#/components/schemas/LifecycleEvent"
+          }
+        },
+        "examples": [
+          {
+            "name": "Default Example",
+            "params": [],
+            "result": {
+              "name": "Default Result",
+              "value": {
+                "state": "hibernated",
+                "previous": "suspended"
+              }
+            }
+          }
+        ]
+      },
+      {
+        "name": "onRestore",
+        "tags": [
+          {
+            "name": "event"
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the restoration event",
+        "params": [],
+        "result": {
+          "name": "value",
+          "schema": {
+            "$ref": "#/components/schemas/LifecycleEvent"
+          }
+        },
+        "examples": [
+          {
+            "name": "Default Example",
+            "params": [],
+            "result": {
+              "name": "Default Result",
+              "value": {
+                "state": "suspended",
+                "previous": "hibernated"
+              }
+            }
+          }
+        ]
+      },
+      {
+        "name": "onDestroy",
+        "tags": [
+          {
+            "name": "event"
+          },
+          {
+            "name": "capabilities",
+            "x-uses": [
+              "xrn:firebolt:capability:lifecycle:state"
+            ]
+          }
+        ],
+        "summary": "Listen to the destroy event",
```

Comment:
We should note in the summary that this is not a reliable event--we can't guarantee that the app will always get this notification in the event the platform needs to quickly kill the app.